src/query-builder/transformer/RawSqlResultsToEntityTransformer.ts

```diff
@@ -46,21 +61,39 @@ export class RawSqlResultsToEntityTransformer {
     transform(rawResults: any[], alias: Alias): any[] {
         const group = this.group(rawResults, alias)
         const entities: any[] = []
-        group.forEach((results) => {
+        for (const results of group.values()) {
```

Comment:
What about using filter and not having to use push ?

Reply:
This is not just a filter, it's a map and a filter, so we do both in one shot.